src/views/Play/Play.tsx

```diff
@@ -729,8 +813,270 @@ export class Play extends React.Component<PlayProperties, any> {
         </div>;
     }
 
+    rengoListHeaders() {
+        return <div className="challenge-row">
+            <span className="head"></span>
+            <span className="head">{_("Organiser")}</span>
+            {/* <span className="head">{_("Rank")}</span> */}
+            <span className="head">{_("Size")}</span>
+            <span className="head time-control-header">{_("Time")}</span>
+            <span className="head">{_("")}</span>
+            <span className="head">{_("Handicap")}</span>
+            <span className="head" style={{textAlign: "left"}}>{_("Name")}</span>
+            <span className="head" style={{textAlign: "left"}}>{_("Rules")}</span>
+        </div>;
+    }
+
+    nominateForRengoChallenge = (C) => {
+        swal({
+            text: _("Joining..."),   // translator: the server is processing their request to join a rengo game
+            type: "info",
+            showCancelButton: false,
+            showConfirmButton: false,
+            allowEscapeKey: false,
+        }).catch(swal.noop);
+
+        put("challenges/%%/join", C.challenge_id, {})
+        .then(() => {
+            swal.close();
+        })
+        .catch((err) => {
+            swal.close();
+            errorAlerter(err);
+        });
+    };
+
+    unNominateForRengoChallenge = (C) => {
+        swal({
+            text: _("Withdrawing..."),   // translator: the server is processing their request to withdraw from a rengo challenge
+            type: "info",
+            showCancelButton: false,
+            showConfirmButton: false,
+            allowEscapeKey: false,
+        }).catch(swal.noop);
+
+        del("challenges/%%/join", C.challenge_id, {})
+        .then(() => {
+            swal.close();
+        })
+        .catch((err) => {
+            swal.close();
+            errorAlerter(err);
+        });
+    };
+
+    rengoList = () => {
```

Comment:
This seems *very* similar to the existing `challengeList()` does it really need to be its own function?

Or at least some duplicate parts be moved into a common function?  Pulling out `cheatAlert(C)` alone would save ~25 lines of JSX.

Reply:
> save ~25 lines of JSX.

/FeelsGoodMan

Reply:
I'm not sure it's worth optimising this yet - it hasn't even been tried for real by anyone other than me.

People may hate having Rengo separately, for example...